### PR TITLE
Fixes from the test party.

### PR DESCRIPTION
**Pull request by nedrocks**

### CODE REVIEW

File: lib/bloc/discussion_list/discussion_list_bloc.dart [hunk 1]
```diff
@@ -228,16 +228,21 @@ class DiscussionListBloc
...
-          if (Random().nextInt(2) == 0) {
-            throw "some random error";
-          }
-          var updatedDiscussion = event.discussion.copyWith(
-            mutedUntil: DateTime.now().add(
-              Duration(seconds: event.mutedForSeconds),
-            ),
-          );
-          this.add(_DiscussionListMuteAsyncSuccessEvent(updatedDiscussion));
+          // await Future.delayed(Duration(seconds: 2));
```
Comment (jasondellaluce) [inline lib/bloc/discussion_list/discussion_list_bloc.dart:231]: Ditto: Remove comments

File: lib/bloc/discussion_list/discussion_list_bloc.dart [hunk 2]
```diff
@@ -228,16 +228,21 @@ class DiscussionListBloc
...
+          //   mutedUntil: DateTime.now().add(
+          //     Duration(seconds: event.mutedForSeconds),
+          //   ),
+          // );
+          // this.add(_DiscussionListMuteAsyncSuccessEvent(updatedDiscussion));
+          await this.repository.updateDiscussionUserSettings(
+              event.discussion.id,
+              null,
+              DiscussionUserNotificationSetting.NONE);
+          this.add(_DiscussionListDeleteAsyncSuccessEvent(event.discussion));
```
Comment (jasondellaluce) [inline lib/bloc/discussion_list/discussion_list_bloc.dart:245]: 2 things here:
- The handler for this success event should be edited to update the local copy of the discussion with the new notification setting status. I took for granted that a Discussion could be returned from this repository call (once available), but if this is not the case we should update it manually.
- The Discussion class should be edited to include the `meNotificationSettings` field, and the fragment should include it too in order to fetch it. In the Discussion class, the `mutedUntil` field should be deleted, as it was a placeholder I've put before the logic on the backend was designed. Same for the `Discussion.isMuted` getter.

File: lib/main.dart [hunk 3]
```diff
@@ -25,7 +25,7 @@ void main() {
       statusBarBrightness: Brightness.dark,
     ),
   );
-  Constants.setEnvironment(Environment.STAGING);
+  Constants.setEnvironment(Environment.DEV);
```
[the remark this answers is not in the document]
Comment (nedrocks) [reply lib/main.dart:28]: Good catch. REverting.

File: lib/bloc/discussion_list/discussion_list_bloc.dart (hunk 2 above)
Comment (nedrocks) [reply lib/bloc/discussion_list/discussion_list_bloc.dart:245]: Updated this. I did not add the isMuted though.

**Pull request merged**